### ISSUE TITLE
[Bug] Failed to set generic field

### ISSUE THREAD

**Issue by onacit** [jqno/equalsverifier#983]
### Describe the bug

With following class,

```java
@Setter
@Getter
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@SuppressWarnings({
        "java:S119" // <SELF ...>
})
public abstract class AbstractResponseType<SELF extends AbstractResponseType<SELF>>
        extends AbstractType<SELF> {

    @SuppressWarnings({"unchecked"})
    public final SELF get() {
        if (wrapped != null) {
            return wrapped;
        }
        return (SELF) this;
    }

    @Valid
    @XmlTransient
    private SELF wrapped; // <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
}
```

EqualsVerifier fails setting field value.

```

java.lang.AssertionError: EqualsVerifier found a problem in class com.github.jinahya.epost.openapi.proxy.cloud.gateway.route._download_area_code_service.AreaCodeInfoResponse.
-> Reflection error: perhaps a ClassLoader problem?
Try re-running with #withResetCaches()

For more information, go to: https://www.jqno.nl/equalsverifier/errormessages
(EqualsVerifier 3.16.1, JDK 21.0.4 on Mac OS X)

	at nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.verify(SingleTypeEqualsVerifierApi.java:351)
	at com.github.jinahya.epost.openapi.proxy.cloud.gateway.route.__common.AbstractTypeTest.__equals(AbstractTypeTest.java:117)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: nl.jqno.equalsverifier.internal.exceptions.ReflectionException: Reflection error: perhaps a ClassLoader problem?
Try re-running with #withResetCaches()
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.wrappedChange(FieldModifier.java:143)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.lambda$change$3(FieldModifier.java:135)
	at nl.jqno.equalsverifier.internal.util.Rethrow.lambda$rethrow$1(Rethrow.java:51)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:34)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:49)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:59)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.change(FieldModifier.java:135)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.changeField(FieldModifier.java:122)
	at nl.jqno.equalsverifier.internal.reflection.InPlaceObjectAccessor.scrambleInternal(InPlaceObjectAccessor.java:80)
	at nl.jqno.equalsverifier.internal.reflection.InPlaceObjectAccessor.scramble(InPlaceObjectAccessor.java:58)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedAccessor(ClassAccessor.java:225)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedObject(ClassAccessor.java:199)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.giveInstances(FallbackFactory.java:99)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.createValues(FallbackFactory.java:40)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.createTuple(PrefabValues.java:189)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.realizeCacheFor(PrefabValues.java:172)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.giveTuple(PrefabValues.java:93)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.giveOther(PrefabValues.java:131)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.lambda$changeField$2(FieldModifier.java:119)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.wrappedChange(FieldModifier.java:140)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.lambda$change$3(FieldModifier.java:135)
	at nl.jqno.equalsverifier.internal.util.Rethrow.lambda$rethrow$1(Rethrow.java:51)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:34)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:49)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:59)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.change(FieldModifier.java:135)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.changeField(FieldModifier.java:122)
	at nl.jqno.equalsverifier.internal.reflection.InPlaceObjectAccessor.scrambleInternal(InPlaceObjectAccessor.java:80)
	at nl.jqno.equalsverifier.internal.reflection.InPlaceObjectAccessor.scramble(InPlaceObjectAccessor.java:58)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedAccessor(ClassAccessor.java:225)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedAccessor(ClassAccessor.java:210)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedObject(ClassAccessor.java:186)
	at nl.jqno.equalsverifier.internal.util.Configuration.ensureUnequalExamples(Configuration.java:219)
	at nl.jqno.equalsverifier.internal.util.Configuration.build(Configuration.java:114)
	at nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.buildConfig(SingleTypeEqualsVerifierApi.java:424)
	at nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.performVerification(SingleTypeEqualsVerifierApi.java:410)
	at nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.verify(SingleTypeEqualsVerifierApi.java:347)
	... 4 more
Caused by: java.lang.IllegalArgumentException: Can not set com.github.jinahya.epost.openapi.proxy.cloud.gateway.route.__common.AbstractResponseType field com.github.jinahya.epost.openapi.proxy.cloud.gateway.route.__common.AbstractResponseType.wrapped to java.lang.Object
	at java.base/java.lang.reflect.Field.set(Field.java:836)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.lambda$changeField$2(FieldModifier.java:120)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.wrappedChange(FieldModifier.java:140)
	... 40 more

Disconnected from the target VM, address: '127.0.0.1:61370', transport: 'socket'

Process finished with exit code 255
```

### Steps to reproduce

* Define a field of a generics type parameter
* Try to verify

### Error message and version number

```

java.lang.AssertionError: EqualsVerifier found a problem in class com.github.jinahya.epost.openapi.proxy.cloud.gateway.route._download_area_code_service.AreaCodeInfoResponse.
-> Reflection error: perhaps a ClassLoader problem?
Try re-running with #withResetCaches()

For more information, go to: https://www.jqno.nl/equalsverifier/errormessages
(EqualsVerifier 3.16.1, JDK 21.0.4 on Mac OS X)

	at nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.verify(SingleTypeEqualsVerifierApi.java:351)
	at com.github.jinahya.epost.openapi.proxy.cloud.gateway.route.__common.AbstractTypeTest.__equals(AbstractTypeTest.java:117)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: nl.jqno.equalsverifier.internal.exceptions.ReflectionException: Reflection error: perhaps a ClassLoader problem?
Try re-running with #withResetCaches()
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.wrappedChange(FieldModifier.java:143)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.lambda$change$3(FieldModifier.java:135)
	at nl.jqno.equalsverifier.internal.util.Rethrow.lambda$rethrow$1(Rethrow.java:51)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:34)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:49)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:59)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.change(FieldModifier.java:135)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.changeField(FieldModifier.java:122)
	at nl.jqno.equalsverifier.internal.reflection.InPlaceObjectAccessor.scrambleInternal(InPlaceObjectAccessor.java:80)
	at nl.jqno.equalsverifier.internal.reflection.InPlaceObjectAccessor.scramble(InPlaceObjectAccessor.java:58)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedAccessor(ClassAccessor.java:225)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedObject(ClassAccessor.java:199)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.giveInstances(FallbackFactory.java:99)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.createValues(FallbackFactory.java:40)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.createTuple(PrefabValues.java:189)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.realizeCacheFor(PrefabValues.java:172)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.giveTuple(PrefabValues.java:93)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.giveOther(PrefabValues.java:131)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.lambda$changeField$2(FieldModifier.java:119)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.wrappedChange(FieldModifier.java:140)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.lambda$change$3(FieldModifier.java:135)
	at nl.jqno.equalsverifier.internal.util.Rethrow.lambda$rethrow$1(Rethrow.java:51)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:34)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:49)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:59)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.change(FieldModifier.java:135)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.changeField(FieldModifier.java:122)
	at nl.jqno.equalsverifier.internal.reflection.InPlaceObjectAccessor.scrambleInternal(InPlaceObjectAccessor.java:80)
	at nl.jqno.equalsverifier.internal.reflection.InPlaceObjectAccessor.scramble(InPlaceObjectAccessor.java:58)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedAccessor(ClassAccessor.java:225)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedAccessor(ClassAccessor.java:210)
	at nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getRedObject(ClassAccessor.java:186)
	at nl.jqno.equalsverifier.internal.util.Configuration.ensureUnequalExamples(Configuration.java:219)
	at nl.jqno.equalsverifier.internal.util.Configuration.build(Configuration.java:114)
	at nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.buildConfig(SingleTypeEqualsVerifierApi.java:424)
	at nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.performVerification(SingleTypeEqualsVerifierApi.java:410)
	at nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.verify(SingleTypeEqualsVerifierApi.java:347)
	... 4 more
Caused by: java.lang.IllegalArgumentException: Can not set com.github.jinahya.epost.openapi.proxy.cloud.gateway.route.__common.AbstractResponseType field com.github.jinahya.epost.openapi.proxy.cloud.gateway.route.__common.AbstractResponseType.wrapped to java.lang.Object
	at java.base/java.lang.reflect.Field.set(Field.java:836)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.lambda$changeField$2(FieldModifier.java:120)
	at nl.jqno.equalsverifier.internal.reflection.FieldModifier.wrappedChange(FieldModifier.java:140)
	... 40 more

Disconnected from the target VM, address: '127.0.0.1:61370', transport: 'socket'

Process finished with exit code 255
```

### Code: EqualsVerifier invocation

https://github.com/jinahya/epost-openapi-proxy/blob/507ddc1762a5631065d971f893b5139097109f12/src/test/java/com/github/jinahya/epost/openapi/proxy/cloud/gateway/route/__common/AbstractTypeTest.java#L114

**Comment by onacit** [comment]
I just created a branch for this issue.

https://github.com/jinahya/epost-openapi-proxy/blob/equals-verifier/src/test/java/com/github/jinahya/epost/openapi/proxy/cloud/gateway/route/__common/AbstractTypeTest.java#L114

**Comment by jqno** [comment]
I see that you closed the issue, was that intentional?
If so, could you let me know how you solved it? I'm curious :slightly_smiling_face:

**Comment by onacit** [comment]
@jqno HI. Sorry for that closing. Honestly, I wan't sure what I was issuing about. The problem is not solved yet. Let me look more what I've done wrong, if any.

**Comment by jqno** [comment]
Don't worry; let me know if you find something

**Comment by onacit** [comment]
@jqno I think I managed to solve my problem, but, I'm not sure I can explain what was wrong.

Let me just share my codes.

Regards.

```java
public abstract class AbstractType<SELF extends AbstractType<SELF>>
        implements Serializable {
}
```

```java
public abstract class AbstractResponseType<SELF extends AbstractResponseType<SELF>>
        extends AbstractType<SELF> {
    private SELF wrapped;
}
```

```java
public abstract class AbstractTypeTest<TYPE extends AbstractType<TYPE>> {

    @DisplayName("equals")
    @Test
    void __equals() {
        __equals(
                EqualsVerifier.simple().forClass(typeClass)
        ).verify();
    }

    protected SingleTypeEqualsVerifierApi<TYPE> __equals(final SingleTypeEqualsVerifierApi<TYPE> verifierApi) {
        return verifierApi
                .withResetCaches()
                .withIgnoredFields(
                        "unknownAttributes",
                        "unknownElements",
                        "unknownProperties"
                )
                .withPrefabValues(
                        List.class,
                        List.of(new Object()),
                        List.of(new Object())
                )
                .withPrefabValues(
                        Map.class,
                        Map.of("a", new Object()),
                        Map.of("b", new Object())
                )
                ;
    }

    protected final Class<TYPE> typeClass; // <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
}
```

```java
public abstract class AbstractResponseTypeTest<TYPE extends AbstractResponseType<TYPE>>
        extends AbstractTypeTest<TYPE> {

    // -----------------------------------------------------------------------------------------------------------------
    @Override
    protected SingleTypeEqualsVerifierApi<TYPE> __equals(final SingleTypeEqualsVerifierApi<TYPE> verifierApi) {
        final var instance1 = newTypeInstance();
        final var instance2 = newTypeInstance();
        instance1.setCmmMsgHeader(new CmmMsgHeader().requestMsgId("a"));
        instance2.setCmmMsgHeader(new CmmMsgHeader().requestMsgId("b"));
        assert !instance2.equals(instance1);
        return super.__equals(verifierApi)
                .withPrefabValues(
                        CmmMsgHeader.class,
                        new CmmMsgHeader().requestMsgId("a"),
                        new CmmMsgHeader().requestMsgId("b")
                )
//                .withPrefabValues(typeClass, instance1, instance2) // <<<<<<<<<<<<<<<<<<<<<<<<<<
                .withPrefabValues(Object.class, instance1, instance2) // <<<<<<<<<<<<<<<<<<<<<<<<<<
                ;
    }
}
```

**Comment by jqno** [comment]
I was able to reproduce your issue. You can work around it by adding prefab values for AbstractReponseType:

```java
EqualsVerifier
        .forClass(AbstractResponseType.class)
        .withPrefabValues(AbstractResponseType.class, ..., ...)
        .verify();
```

I don't understand (yet) why you're seeing the error that you see; it was intended for a Quarkus edge case which doesn't apply here. I'll investigate that further.

**Comment by jqno** [comment]
I just released version 3.16.2, which improves the error message in your situation.
